src/Psalm/CallMap.php

```diff
@@ -2506,7 +2506,7 @@
 'getmypid' => ['int'],
 'getmyuid' => ['int'],
 'get_object_vars' => ['array', 'obj'=>'object'],
-'getopt' => ['string[]|false[]|array[]', 'options'=>'string', 'longopts='=>'array', '&w_optind='=>'int'],
+'getopt' => ['array<string,string>|array<string,false>|array<string,array<int,string|false>>', 'options'=>'string', 'longopts='=>'array', '&w_optind='=>'int'],
```

Comment:
Do you want the more compact `<array,string|false|array<int,string|false>>` here?

Reply:
I could, I didn't know if psalm supported nested union types in array types (In CallMap.php) yet. (That would be the first occurence)

Reply:
Everything just goes into `Type::parseString` – if it can handle it in a docblock, it can handle it here. That also goes for `get_defined_functions`, which could use `array{user:...,internal:...}` instead of just a generic array.

Reply:
Fixed. Also, missed that I was using nested array types already.  A simple test of get_defined_functions() works properly.